### PR TITLE
Give `go test` and extensive logging example.

### PR DESCRIPTION
**Pull request by hh**

### CODE REVIEW

File: org/godog.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+	ctx.BeforeScenario(func(sc *godog.Scenario) {
+		log.Printf("We are BeforeScenario : %#v : %#v : %#v\n",
+			sc.Uri,  // "example.feature"
+			sc.Name, // ""
+			sc.Id,   // "7"
+		)
+		log.Printf("Settings Godogs = 0")
+		// ./example_test.go:93:3: undefined: Godogs
+		// How do we set godocs?
+	})
```
Comment (hh) [inline org/godog.md:127]: Maybe this is where we could look at creating / setting up the test-target?

File: org/godog.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	ctx.BeforeStep(func(st *godog.Step) {
+		log.Printf("We are BeforeStep : %#v : %#v\n", st.Id, st.Text)
+	})
+	ctx.AfterStep(func(st *godog.Step, err error) {
+		log.Printf("We are AfterStep : %#v : %#v\n", st.Id, st.Text)
+		if err != nil {
+			log.Printf("AfterStep err : %#v", err)
+			// log.Printf("AfterStep err : %#v", err.errorString)
+		}
+	})
```
Comment (hh) [inline org/godog.md:148]: The AfterStep and AfterScenario both have the error passed in.

File: org/godog.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+```example
+
+```
+
+<a id="org7112e79"></a>
+
+# Go Dog (on evolving file)
+
+```shell
+~/go/bin/godog example.feature
```
Comment (hh) [inline org/godog.md:195]: Running as usual with the `godog` command with our familiar output.

File: org/godog.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    1.29725ms
+
+<a id="orgbfb115b"></a>
+
+# Go test
+
+Running go test instead!
+
+```shell
+go test
```
Comment (hh) [inline org/godog.md:238]: Or now also with `go test`

**Review by hh — COMMENTED** [review]
This example may help in giving us the right hooks into the suite/scenario/step.
I think go test may also be able to create a stand alone binary, which seems to be a requirement.

**Comment by zachmandeville** [comment]
@hh, I like these examples, and the combo of go test and an xml output is quite cool.  Could we reduce the amount of committed files though? Specifically, I am wondering if we need:
- junit.xml
- go_logging.md
- features/example.feature

the xml file is generated by executing the org file, so likely does not need to be tracked.  The feature file is identical to the one in our org folder, and my preference is to only have one of these, as it can be disorienting navigating the repo otherwise.

For the markdown file, are there things we get in that export that we don't get in the org file?  

Thank you for this pr!

**Comment by hh** [comment]
@zz I squashed/rebased your requested changes into a single commit.

**Comment by zachmandeville** [comment]
love it!

**Pull request merged**